core/src/avm1/globals/rectangle.rs

```diff
@@ -0,0 +1,1156 @@
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let x = this
+        .get("x", avm, context)?
+        .resolve(avm, context)?
+        .as_number(avm, context)?;
+    let y = this
+        .get("y", avm, context)?
+        .resolve(avm, context)?
+        .as_number(avm, context)?;
+    let (horizontal, vertical) = value_to_point(
+        args.get(0).unwrap_or(&Value::Undefined).to_owned(),
+        avm,
+        context,
+    )?;
+
+    this.set("x", Value::Number(x + horizontal), avm, context)?;
+    this.set("y", Value::Number(y + vertical), avm, context)?;
+
+    Ok(Value::Undefined.into())
+}
+
+fn intersection<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let this_left = this
+        .get("x", avm, context)?
+        .resolve(avm, context)?
+        .as_number(avm, context)?;
+    let this_top = this
+        .get("y", avm, context)?
+        .resolve(avm, context)?
+        .as_number(avm, context)?;
+    let this_right = this_left
+        + this
+            .get("width", avm, context)?
+            .resolve(avm, context)?
+            .as_number(avm, context)?;
+    let this_bottom = this_top
+        + this
+            .get("height", avm, context)?
+            .resolve(avm, context)?
+            .as_number(avm, context)?;
+
+    let (other_left, other_top, other_width, other_height) =
+        if let Some(Value::Object(other)) = args.get(0) {
+            (
+                other
+                    .get("x", avm, context)?
+                    .resolve(avm, context)?
+                    .as_number(avm, context)?,
+                other
+                    .get("y", avm, context)?
+                    .resolve(avm, context)?
+                    .as_number(avm, context)?,
+                other
+                    .get("width", avm, context)?
+                    .resolve(avm, context)?
+                    .as_number(avm, context)?,
+                other
+                    .get("height", avm, context)?
+                    .resolve(avm, context)?
+                    .as_number(avm, context)?,
+            )
+        } else {
+            (NAN, NAN, NAN, NAN)
+        };
+    let other_right = other_left + other_width;
+    let other_bottom = other_top + other_height;
+
+    let (left, top, right, bottom) = if this_left.is_nan()
+        || other_left.is_nan()
+        || this_top.is_nan()
+        || other_top.is_nan()
+        || this_right.is_nan()
+        || other_right.is_nan()
+        || this_bottom.is_nan()
+        || other_bottom.is_nan()
+    {
+        (0.0, 0.0, 0.0, 0.0)
+    } else {
+        (
+            this_left.max(other_left),
+            this_top.max(other_top),
+            this_right.min(other_right),
+            this_bottom.min(other_bottom),
+        )
```

Comment:
Also should return (0, 0, 0, 0) when min >= max:
```
var rect1 = new Rectangle(1, 1, 1, 1);
var rect2 = new Rectangle(3, 3, 1, 1);
trace(rect1.intersection(rect2));
```

Reply:
👍